File: tests/fixtures/functions/all_forms.py
# 1. Simple function
def greet(name: str) -> str:
    return f"Hello, {name}"

# 2. Function with default parameter
def repeat(text: str, times: int = 1) -> str:
    return text * times

# 3. Async function
async def fetch_data(url: str) -> dict:
    return {}

# 4. Method in class
class Calculator:
    def add(self, a: int, b: int) -> int:
        return a + b

    def get_value(self) -> int:
        return 0

File: tests/fixtures/functions/all_forms.ts
// 1. Function declaration
function greet(name: string): string {
    return `Hello, ${name}`;
}

// 2. Generator function
function* range(start: number, end: number): Generator<number> {
    for (let i = start; i < end; i++) {
        yield i;
    }
}

// 3. Arrow function
const double = (x: number): number => x * 2;

// 4. Async function
async function fetchData(url: string): Promise<string> {
    const response = await fetch(url);
    return response.text();
}

// 5. Exported function
export function add(a: number, b: number): number {
    return a + b;
}

// 6. Function expression
const square = function(x: number): number {
    return x * x;
};

// 7. Method in class
class MathUtil {
    multiply(a: number, b: number): number {
        return a * b;
    }

    static identity<T>(x: T): T {
        return x;
    }
}
